### ISSUE TITLE
Faker starter seed

### ISSUE THREAD

**Issue by tsdevelopment** [zenstruck/foundry#284]
I try to set the faker seed to get the same data when creating new fake data. I have seen that there is already an issue:
https://github.com/zenstruck/foundry/issues/128

But the provided solutions don't work for me. I tried to seht my own generator service. In my generator class everything seems to work, but my factory produces always different results.

```yaml
# zenstruck_foundry.yaml
zenstruck_foundry:
    auto_refresh_proxies: true
    faker:
        service: App\Service\Faker\Generator
```

```php
// service
<?php

namespace App\Service\Faker;

use Faker\Factory;
use Faker\Generator as FakerGenerator;

final class Generator extends FakerGenerator
{
    public function __construct()
    {
        $faker = Factory::create();
        $faker->seed(4920);

        foreach ($faker->getProviders() as $provider) {
            $this->addProvider($provider);
        }

        echo $faker->email() // --> WORKS: Produces the same result on each run.

        return $faker;
    }
}
```

But the result in my factory is always different:

```php
<?php

namespace App\Factory;

use Zenstruck\Foundry\ModelFactory;

final class MyFactory extends ModelFactory
{
    function getDefaults(): array
    {
        return [
            'email' => self::faker()->email(), // --> DOESN'T WORK: Should always produce the same result on each run
        ];
    }

    protected static function getClass(): string
    {
        return MyClass::class;
    }
}

```

Is there any example to set the seed for all factories?

**Comment by kbond** [comment]
Hey @tsdevelopment, thanks for bringing this to my attention. Indeed it is very unclear how to do this.

Right now, this is the way to do what you want:

```yaml
# zenstruck_foundry.yaml

services:
    app_faker:
        class: Faker\Generator
        factory: [Faker\Factory, 'create']
        calls:
            - seed: [4920]

zenstruck_foundry:
    auto_refresh_proxies: true
    faker:
        service: app_faker
```

_That's it, no custom class required._

I am going to improve the docs to make this clearer. Also, seeing as setting a seed is somewhat common, I'm going to add a config option for this:

```yaml
# zenstruck_foundry.yaml
zenstruck_foundry:
    auto_refresh_proxies: true
    faker:
        seed: 4920 # does not yet exist
```

Look for this in a future release!

**Comment by kbond** [comment]
FYI, the seed config option is available in [v1.21.0](https://github.com/zenstruck/foundry/releases/tag/v1.21.0).

**Comment by tsdevelopment** [comment]
That works. Thank you for your support!
